src/foam/dao/index/ScanPlan.java

```diff
@@ -56,7 +56,7 @@ public long calculateCost(PropertyInfo propertyInfo) {
         order_   = null;
       } else {
         sortRequired = true;
-        cost *= Math.log(cost) / Math.log(2);
+        cost = (long) (Math.log(cost) / Math.log(2));
```

Comment:
Why this change? Isn't the sorting cost n * log2(n)?

Reply:
Is it a search instead of sort? When searching the indexed plan this cost is being used.

Reply:
But we said sortRequired = true;
So we will need to sort the values first.

Reply:
Fixed in https://github.com/foam-framework/foam2/pull/3791/commits/00294323a4d50cc6a268b2564b4edcd352ddc90e.